gsa/src/web/pages/tickets/component.js

```diff
@@ -0,0 +1,281 @@
+  }
+
+  handleUserIdChange(userId) {
+    this.setState({userId});
+  }
+
+  handleInteraction() {
+    const {onInteraction} = this.props;
+    if (isDefined(onInteraction)) {
+      onInteraction();
+    }
+  }
+
+  handleSolve({
+    ticketId,
+    comment,
+  }) {
+    const {
+      gmp,
+      onSolved,
+      onSolveError,
+    } = this.props;
+
+    this.setState({solvedDialogVisible: false});
+
+    return gmp.ticket.solve({id: ticketId, comment})
+      .then(onSolved, onSolveError);
+  }
+
+  handleClose({
+    ticketId,
+    comment,
+  }) {
+    const {
+      gmp,
+      onClosed,
+      onCloseError,
+    } = this.props;
+
+    this.setState({closedDialogVisible: false});
+
+    return gmp.ticket.close({id: ticketId, comment})
+      .then(onClosed, onCloseError);
+  }
+
+  render() {
+    const {
+      users,
+      children,
+      onCloned,
+      onCloneError,
+      onCreated,
+      onCreateError,
+      onDeleted,
+      onDeleteError,
+      onDownloaded,
+      onDownloadError,
+      onInteraction,
+      onSaved,
+      onSaveError,
+    } = this.props;
+    const {
+      closedDialogVisible,
+      createDialogVisible,
+      result,
+      solvedDialogVisible,
+      ticket,
+      userId,
+    } = this.state;
+    return (
+      <EntityComponent
+        name="ticket"
+        onCreated={onCreated}
+        onCreateError={onCreateError}
+        onCloned={onCloned}
+        onCloneError={onCloneError}
+        onDeleted={onDeleted}
+        onDeleteError={onDeleteError}
+        onDownloaded={onDownloaded}
+        onDownloadError={onDownloadError}
+        onInteraction={onInteraction}
+        onSaved={onSaved}
+        onSaveError={onSaveError}
+      >
+        {({
+          create,
+          save,
+          ...other
+        }) => (
+          <React.Fragment>
+            {children({
+              ...other,
+              createFromResult: this.handleOpenCreateDialog,
+              solve: this.handleOpenSolvedDialog,
+              close: this.handleOpenClosedDialog,
+            })}
+            {createDialogVisible &&
+              <CreateTicketDialog
+                resultId={result.id}
+                title={_('Create new Ticket for Result {{name}}', result)}
+                userId={isDefined(userId) ? userId : selectSaveId(users)}
+                users={users}
+                onUserIdChange={this.handleUserIdChange}
+                onClose={this.handleCloseCreateDialog}
+                onSave={d => {
+                  this.handleInteraction();
+                  return save(d).then(this.handleCloseCreateDialog);
+                }}
+              />
+            }
+            {solvedDialogVisible &&
+              <StatusChangeTicketDialog
+                title={_('Mark Ticket {{name}} as solved', ticket)}
+                ticketId={ticket.id}
+                onClose={this.handleCloseSolvedDialog}
+                onSave={this.handleSolve}
+              />
```

Comment:
```handleSolve``` and ```handleClose``` don't call ```this.handleInteraction()```. Does a gmp command do that somehow? If not, it would be nice for these two methods to also be regarded as session prolonging action just as ```onSave``` of the ```CreateTicketDialog``` does.